### ISSUE TITLE
how to run code server in https

### ISSUE THREAD

**Issue by sanjayme97** [cdr/code-server#1548]
if we run  _"docker run -d -p 8080:8080  sanjayme/vscode-dotnet-core --auth none --cert"_ still we are facing not secure connection how to solve this. even we tried something similar to _"docker -v /path/to/certs.crt:/home/coder/cert.crt -v /path/to/cert-key.key:/home/coder/cert.key -p 8080:8080 run codercom/code-server:v2 --cert /home/coder/cert.crt --cert-key /home/coder/cert.key"_  @code-asher kindly help here.
**http url is working but if we browse https dns mapped url its throwing error. check the attachment for error**
![image](https://user-images.githubusercontent.com/45826938/80099471-258af200-858c-11ea-8165-5df41183c9d9.png)

**Comment by mahmoudajawad** [comment]
I would suggest you run `code-server` as it's on separate container, then run a `nginx` reverse-proxy container pointing at `code-server` port. That allows to manage your reverse-proxy without the need to reset/restart/rebuild your `code-server` container. This can be easily achieved with docker-compose with both `code-server` and `nginx` as services in one compose config.

**Comment by code-asher** [comment]
I ran your image but it worked for me. What version of Chrome are you running? I'm on 80. I wonder if there was a recent change that makes self-signed certificates not work for websockets.

**Comment by sanjayme97** [comment]
@code-asher I am on version 81 . is there any other way than self-signed certificates  to run it in https in server?

**Comment by sanjayme97** [comment]
> I would suggest you run `code-server` as it's on separate container, then run a `nginx` reverse-proxy container pointing at `code-server` port. That allows to manage your reverse-proxy without the need to reset/restart/rebuild your `code-server` container. This can be easily achieved with docker-compose with both `code-server` and `nginx` as services in one compose config.

**Comment by sanjayme97** [comment]
@mahmoudajawad I am not getting it. is there any reference/example can you share it.

**Comment by sanjayme97** [comment]
@code-asher if I run `"docker run -d -p 8080:8080 sanjayme/vscode-dotnet-core --auth none --cert"` i also can browse https://8080 but not secure
![image](https://user-images.githubusercontent.com/45826938/80173327-1009dc80-860d-11ea-8e7d-a215b069e89d.png)

and **below error is when we browse dns url with ssl attached** 
![error](https://user-images.githubusercontent.com/45826938/80173377-3af43080-860d-11ea-80bf-f57a7c138970.PNG)

**Comment by code-asher** [comment]
Ah that's interesting. Are you using some kind of reverse proxy? The websocket connection is returning a 200 which is unusual.

**Comment by necmettin** [comment]
I have a similar setup, tried nginx reverse proxy but couldn't get code-server to accept the domain name no matter what I did. (I use cloudflare and keep the origin server non-https). So I switched to a self-signed certificate on non-80 port.

Long story short, Chrome 81 does not let you pass self-signed certificates. Firefox does. code-server wasn't running on Chrome but when I switched to Firefox, no problems.
